src/core/store/modules/wallet/wallet.selectors.ts

```diff
@@ -2,7 +2,7 @@ import { getNetwork } from '@utils';
 import { RootState } from '@types';
 
 const selectWallet = (state: RootState) => state.wallet;
-const selectWalletIsConnected = (state: RootState) => !!state.wallet.selectedAddress;
+const selectWalletIsConnected = (state: RootState) => state.wallet.isConnected;
```

Comment:
I remember we changed this a while ago since sometimes onboard remains connected but no longer have a selected address. There where some edge cases where that happened, like when selecting another wallet and then dismissing the prompt. Would suggest we dont change this unless its really necessary, and have someone test it thoroughly.

Reply:
Change was done here, while testing multi chain support, https://github.com/yearn/yearn-finance-v3/commit/7bec8af03e4e7056bc52653581d013b5c3531aac#diff-031294351bc496adc7c6604ac7cdc16dc2a287110e5a14931d9c3e0a7edcbdc2L4

Reply:
should we do `const selectWalletIsConnected = (state: RootState) => state.wallet.isConnected && !!state.wallet.selectedAddress` 

@xgambitox wdyt

Reply:
@xgambitox pushed the logic from the above comment, tested and working fine, but we should have fox test too